src/select_from_weighted.rs

```diff
@@ -45,7 +45,7 @@ pub fn select_from_weighted<T: Clone, W: Uint + SampleUniform>(
         }
         total_weight = total_weight
             .checked_add(*weight)
-            .ok_or_else(|| String::from("Total weight is greater than maximum value of u32"))?;
+            .ok_or_else(|| String::from("Total weight is greater than maximum value of u128"))?;
```

Comment:
Nice find. However, the value range is not always u128. It is the range of the generic parameter `W`, which can be u8, u16, u32, u64, u128 or usize. So we need a more generic message here, such as

```suggestion
            .ok_or_else(|| String::from("Total weight is greater than maximum value"))?;
```

It is probably also puble to use `let bits = core::mem::size_of::<W>()*8` and add this to the error message but it feels a bit overkill here.

Reply:
Yeah, you are right. It is better to use a generic message, since it depends from the generic parameter `W`.
Moreover, I fully agree with you about the fact that the usage of `sizeof` here for such string is an overkill 😂.